api/src/utils/send-request.ts

```diff
@@ -0,0 +1,92 @@
+import axios from 'axios';
+import env from '../env';
+import { RequestConfig, RequestProxyConfig } from '../types';
+import { getConfigFromEnv } from './get-config-from-env';
+
+/* The sense behind returning false:
+ * So if you are on a unix environment, you are able to configure your proxy system wide.
+ * But there is no continuity. Sometimes the configuration is in UPPERCASE sometimes in lowercase.
+ * And this really matters for some libs out there.
+ * So false means: Do not use any of the env variables out there. Just do nothing. My system is configured properly.
+ * see: https://about.gitlab.com/blog/2021/01/27/we-need-to-talk-no-proxy/
+ *
+ * Remarks to undici: In undici, we are in a very low level to fetch api.
+ * So there is no "magic" going on. Means that if you want to use an other proxy then systemproxy.
+ * You have to configure it. There is no auto parsing for HTTP_PROXY/http_proxy or something else.
+ *
+ * So - line 20 and 21 could be removed, line 24 should be false. -> use proxy? yes -> config please!
+ */
+function getProxyOptions(): false | RequestProxyConfig | undefined {
+	const useSystemProxy = env.REQUEST_PROXY === false;
+	if (useSystemProxy) return false;
+
+	const hasGlobalProxy = Object.keys(getConfigFromEnv('REQUEST_PROXY_')).length > 0;
+	if (!hasGlobalProxy) return undefined;
+
+	// proxy configuration
+	const proxyProtocol = env.REQUEST_PROXY_PROTOCOL ?? 'http';
+	const proxyHost = env.REQUEST_PROXY_HOST;
+	const proxyPort = env.REQUEST_PROXY_PORT;
+	const proxyAuth = getConfigFromEnv('REQUEST_PROXY_AUTH_');
+	const proxyUsername = env.REQUEST_PROXY_AUTH_USERNAME;
+	const proxyPassword = env.REQUEST_PROXY_AUTH_PASSWORD;
+
+	// proxy validation
+	if (!proxyHost || !proxyPort) {
+		throw new Error(
+			`REQUEST_PROXY_HOST and REQUEST_PROXY_PORT are required, if you specify proxy configuration as an object.`
+		);
+	}
+	if (Object.keys(proxyAuth).length === 1) {
+		throw new Error(
+			`REQUEST_PROXY_AUTH_USERNAME and REQUEST_PROXY_AUTH_PASSWORD are required, if you specify proxy authentication.`
+		);
+	}
```

Comment:
This `if` statement doesn't check the same condition as the error is describing. The condition is whether or not the `REQUEST_PROXY_AUTH` namespace has more (or less!) than one thing in it, not whether the username/password combo exists

Reply:
This if checks if only one from REQUEST_PROXY_AUTH_X is set.. you need both or none... So this check should be fine I think... Either none of them or both of them :)

Reply:
Right now I could set `REQUEST_PROXY_AUTH_USER` and `REQUEST_PROXY_AUTH_PW` and it would validate as correct against this if statement. The number of configured properties isn't the same as configuring the two exact properties you're looking for 🙂

Reply:
Ah - now I got you :)
Yes you are right... I assumed that config keys will be copied correctly from the docs :D

I will change that... But maybe we can do a beta release for this PR? I neeeeeed this very urgent in my environment :) I can't access remote endpoints in flows without these settings :(

Will do update on this PR tomorrow...

Reply:
> I assumed that config keys will be copied correctly from the docs :D

You'd be surprised 😅 

> I will change that... But maybe we can do a beta release for this PR? I neeeeeed this very urgent in my environment :) I can't access remote endpoints in flows without these settings :(

I can't really do a release with an in-works feature that isn't prod ready no.. You can build the API from source and run that as a test though!

Reply:
> I can't really do a release with an in-works feature that isn't prod ready no.. You can build the API from source and run that as a test though! 

Yes I do this already... But keeping directus up2date is a bit tricky, cause I need to merge this all the time back into new version, build, start again... Some extra work but okay - I'll fix this tomorrow :)